src/tests/Microsoft.PowerFx.Core.Tests/IRTests/IRTranslatorTests.cs

```diff
@@ -54,5 +56,19 @@ public void TestLazyEvalNode(string expression, string expectedFragment, Type ty
             // Type Check
             Assert.Equal(type, lazyEvalNode.IRContext.ResultType.GetType());
         }
+
+        [Theory]
+        [InlineData(@"""abc"" = 23")]
+        [InlineData(@"23 = ""abc""")]
+        [InlineData(@"23 <> ""abc""")]
+        [InlineData(@"""abc"" <> 23")]
+        public void ValidateWarningIssuedWhenCoerceNotWorking(string expression)
+        {
+            var engine = new Engine(new PowerFxConfig());
+            var result = engine.Check(expression);
+
+            Assert.True(result.IsSuccess);
+            Assert.Equal(1, result.Errors.Count(x => x.IsWarning));
```

Comment:
This is key for why we need a C# test here - we don't (yet!) have a way to validate warnings in a .txt.